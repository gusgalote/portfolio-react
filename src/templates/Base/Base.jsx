import P from 'prop-types';

import * as Styled from './styles';

const Base = ({ children }) => {
  return (
    <Styled.Main>
      <Styled.Container>{children}</Styled.Container>
    </Styled.Main>
  );
};

Base.propTypes = {
  children: P.node.isRequired,
};

export { Base };
