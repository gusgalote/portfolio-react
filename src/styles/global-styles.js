import { createGlobalStyle, css } from 'styled-components';

const GlobalStyles = createGlobalStyle`
   * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
    scroll-behavior: smooth;

    min-height: 100vh;
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) => theme.font.family.default};

    color: ${({ theme }) => theme.colors.text};
    background: ${({ theme }) => theme.colors.background};
  }

  h1, h2, h3, h4, h5, h6 {
    margin: ${({ theme }) => theme.spacings.large} 0;
  }

  p {
    margin: ${({ theme }) => theme.spacings.medium} 0;
  }

  ul, ol {
    margin: ${({ theme }) => theme.spacings.large} 0;
    padding: ${({ theme }) => theme.spacings.large} 0;
  }

  a {
    color: ${({ theme }) => theme.colors.secondaryColor}
  }

  .table {
    width: 100%;
    overflow-y: auto;
  }
`;

export { GlobalStyles };
