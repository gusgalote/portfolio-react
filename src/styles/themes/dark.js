const dark = {
  colors: {
    primary: '#AA97d2',
    secondary: '#bfd297',
    background: '#121212',
    text: '#fff',
    white: '#fff',
    black: '#0d0d0d',
    grayLighter: '#ededed',
    grayLight: '#e0e0e0',
    gray: '#595959',
    grayDark: '#333',
    grayDarker: '#1a1a1a',
    gradientPurple:
      'linear-gradient(90deg, rgba(170,151,210,1) 35%, rgba(120,99,165,1) 51%, rgba(102,87,139,1) 100%)',
    gradientDark:
      'linear-gradient(184deg, rgba(51,51,51,1) 1%, rgba(34,34,34,1) 99%)',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
    weights: {
      xlight: '200',
      light: '300',
      default: '400',
      semibold: '500',
      bold: '600',
    },
    sizes: {
      xxsmall: '0.8rem',
      xsmall: '1.2rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    lessThanMedium: '(max-width: 768px)',
    moreThanExtraLarge: '(min-width: 1300px)',
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  radius: {
    default: '18px',
    big: '34px',
  },
};

export { dark };
