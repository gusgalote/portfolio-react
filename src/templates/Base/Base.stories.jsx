import { Base } from './Base';

export default {
  title: 'Base',
  component: Base,
};

export const Template = (args) => {
  return (
    <Base {...args}>
      <span style={{ fontSize: '2rem', padding: '0', margin: '0' }}>TEST</span>
    </Base>
  );
};
