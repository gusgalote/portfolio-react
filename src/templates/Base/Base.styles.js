import styled, { css } from 'styled-components';

export const Main = styled.main`
  display: block;
`;

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;

    padding-top: ${theme.spacings.huge};
    padding-bottom: ${theme.spacings.huge};

    padding-right: ${theme.spacings.small};
    padding-left: ${theme.spacings.small};
    margin-right: auto;
    margin-left: auto;

    @media ${theme.media.lessThanMedium} {
      max-width: 540px;
    }

    @media ${theme.media.moreThanExtraLarge} {
      max-width: 1230px;
    }
  `}
`;

export const Row = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-wrap: wrap;
    margin-top: 0;
    margin-right: 2.6rem;
    margin-left: 2.6rem;
  `}
`;
