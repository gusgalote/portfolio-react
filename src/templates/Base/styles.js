import styled, { css } from 'styled-components';

export const Main = styled.main`
  display: block;
`;

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;

    padding-top: ${theme.spacings.xlarge};
    padding-bottom: ${theme.spacings.xlarge};

    padding-right: ${theme.spacings.small};
    padding-left: ${theme.spacings.small};
    margin-right: auto;
    margin-left: auto;

    @media ${theme.media.lessThanMedium} {
      max-width: 540px;
    }

    @media ${theme.media.moreThanExtraLarge} {
      max-width: 1230px;
    }
  `}
`;
