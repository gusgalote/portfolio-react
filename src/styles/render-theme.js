import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { dark as theme } from './themes/dark';

const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

export { renderTheme };
