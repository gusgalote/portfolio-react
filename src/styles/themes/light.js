import { dark } from './dark';

const light = {
  colors: {
    primary: '#7863A5',
    primaryLight: '#a294c1',
    primaryDark: '#554577',
    background: '#f6f8fb',
    text: '#121212',
    white: '#fff',
    black: '#0d0d0d',
    grayLighter: '#ededed',
    grayLight: '#e0e0e0',
    gray: '#595959',
    grayDark: '#333',
    grayDarker: '#1a1a1a',
    gradient:
      'linear-gradient(90deg, rgba(170,151,210,1) 35%, rgba(120,99,165,1) 51%, rgba(102,87,139,1) 100%)',
  },
  ...dark,
};

export { light };
