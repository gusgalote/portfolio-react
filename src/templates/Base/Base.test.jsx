import { screen } from '@testing-library/react';

import { renderTheme } from '../../styles/render-theme';
import { dark as theme } from '../../styles/themes/dark';

import { Base } from '../Base/Base';

describe('<Base />', () => {
  it('should match snapshot for default styles', () => {
    const { container } = renderTheme(<Base>test</Base>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change max-width on breakpoints', () => {
    const { container } = renderTheme(<Base>test</Base>);

    const main = container.querySelector('main');

    const containerToBeTested = main.firstChild;

    expect(containerToBeTested).toHaveStyleRule('max-width', '540px', {
      media: theme.media.lessThanMedium,
    });

    expect(containerToBeTested).toHaveStyleRule('max-width', '1230px', {
      media: theme.media.moreThanExtraLarge,
    });
  });
});
