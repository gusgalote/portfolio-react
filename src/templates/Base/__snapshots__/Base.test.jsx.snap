// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should match snapshot for default styles 1`] = `
.c0 {
  display: block;
}

.c1 {
  width: 100%;
  padding-top: 5.6rem;
  padding-bottom: 5.6rem;
  padding-right: 1.6rem;
  padding-left: 1.6rem;
  margin-right: auto;
  margin-left: auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 0;
  margin-right: 2.6rem;
  margin-left: 2.6rem;
}

@media (max-width:768px) {
  .c1 {
    max-width: 540px;
  }
}

@media (min-width:1300px) {
  .c1 {
    max-width: 1230px;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      test
    </div>
  </div>
</main>
`;
